server/src/main/java/org/opensearch/search/aggregations/metrics/SumAggregator.java

```diff
@@ -148,6 +156,25 @@ public LeafBucketCollector getStarTreeCollector(LeafReaderContext ctx, LeafBucke
         );
     }
 
+    /**
+     * The parent aggregator invokes this method to get a StarTreeBucketCollector,
+     * which exposes collectStarTreeEntry() to be evaluated on filtered star tree entries
+     */
+    public StarTreeBucketCollector getStarTreeBucketCollector(
+        LeafReaderContext ctx,
+        CompositeIndexFieldInfo starTree,
+        StarTreeBucketCollector parentCollector
+    ) throws IOException {
+        return StarTreeQueryHelper.getStarTreeBucketMetricCollector(
+            starTree,
+            MetricStat.SUM.getTypeName(),
+            valuesSource,
+            parentCollector,
+            (bucket) -> sums = context.bigArrays().grow(sums, bucket + 1),
```

Comment:
This needs to be similar to the above method, lets use kahan summation so that precision is maintained - we need to account for both sum and delta

Reply:
Gotcha - making the changes - might have to capture the tests in a follow-up PR specifically for this.